pkg/intel/metadata/manifest/crypto_routines.go

```diff
@@ -41,14 +40,14 @@ const (
 )
 
 var hashInfo = []struct {
-	alg  Algorithm
-	hash hash.Hash
+	alg         Algorithm
+	hashFactory func() hash.Hash
 }{
-	{AlgSHA1, crypto.SHA1.New()},
-	{AlgSHA256, crypto.SHA256.New()},
-	{AlgSHA384, crypto.SHA384.New()},
-	{AlgSHA512, crypto.SHA512.New()},
-	{AlgSM3_256, sm3.New()},
+	{AlgSHA1, crypto.SHA1.New},
+	{AlgSHA256, crypto.SHA256.New},
+	{AlgSHA384, crypto.SHA384.New},
+	{AlgSHA512, crypto.SHA512.New},
+	{AlgSM3_256, sm3.New},
 }
```

Comment:
There was a bug: if I call `Algorithm.Hash()` then I get a hash function which sometimes is already in a dirty state. Fixing.